subsys/jwt/Kconfig

```diff
@@ -7,27 +8,74 @@ menuconfig JWT
 	help
 	  Enable creation of JWT tokens
 
+if JWT
+
 choice
 	prompt "JWT signature algorithm"
 	default JWT_SIGN_RSA
-	depends on JWT
 	help
 	  Select which algorithm to use for signing JWT tokens.
 
 config JWT_SIGN_RSA
 	bool "Use RSA signature (RS-256)"
-	depends on CSPRNG_ENABLED
-	select MBEDTLS
-	select MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
-	select PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY if PSA_CRYPTO_CLIENT
-	select PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_IMPORT if PSA_CRYPTO_CLIENT
 
 config JWT_SIGN_ECDSA
 	bool "Use ECDSA signature (ES-256)"
+
+endchoice
+
+choice
+	default JWT_USE_PSA
+	prompt "Select crypto library to be used"
+
+config JWT_USE_PSA
+	bool "PSA crypto API library"
```

Comment:
Don't you want to add some `depends on` here?

Reply:
Right. I implemented what you proposed [here](https://github.com/zephyrproject-rtos/zephyr/pull/78243#discussion_r1758327347)

Reply:
As this is changing the default behavior (now defaulting to PSA, enabling what's needed) we may want a migration guide entry.